### PR TITLE
Perform a couple of renames on differentiability attribute nodes

### PR DESCRIPTION
**Pull request by ahoppen**
Mostly change parameter to argument where the parameter is actually an argument.

### CODE REVIEW

File: Tests/SwiftParserTest/AttributeTests.swift [hunk 1]
```diff
@@ -51,8 +51,8 @@ final class AttributeTests: XCTestCase {
       diagnostics: [
         DiagnosticSpec(
           locationMarker: "1️⃣",
-          message: "expected ':' and parameters in '@differentiable' argument",
-          fixIts: ["insert ':' and parameters"]
+          message: "expected ':' and arguments in '@differentiable' argument",
+          fixIts: ["insert ':' and arguments"]
```
Comment (bnbarham) [inline Tests/SwiftParserTest/AttributeTests.swift:55]: I guess this doesn't read any worse than before. But... it is a bit odd 😅. The problem is that `Clause` (containing wrt and `Arguments`), `Arguments` (which contains the parentheses), and `ArgumentList` (between the parentheses) all have basically the same name (+- an s).

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/574

@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/574

@swift-ci Please test Windows

File: Tests/SwiftParserTest/AttributeTests.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/SwiftParserTest/AttributeTests.swift:55]: Yes. If there was more use of the `@differentiable` attribute, I would probably agree but for now, I’d suggest we keep it as-is.

**Pull request merged**